src/include/cache.h

```diff
@@ -73,8 +73,10 @@ struct __wt_cache {
 	 * be exact, they can't be garbage, we track what comes in and what goes
 	 * out and calculate the difference as needed.
 	 */
-	uint64_t bytes_dirty;		/* Bytes/pages currently dirty */
-	uint64_t pages_dirty;
+	uint64_t bytes_dirty_intl;	/* Bytes/pages currently dirty */
+	uint64_t pages_dirty_intl;
+	uint64_t bytes_dirty_leaf;
+	uint64_t pages_dirty_leaf;
```

Comment:
Note that I deliberately didn't change the existing stats, but that means there's no stat we can look at to know for sure how much of the dirty content in cache belongs to leaf pages.  Happy to discuss how to address this.